level-0/nfasload.lisp

```diff
@@ -316,11 +316,20 @@
   (dolist (name (pkg.names p))
     (setf (package-ref.pkg (register-package-ref name)) p)))
 
+;; Early definitions. Redefined in l1-symhash.
+(defun package-%local-nicknames (package)      (declare (ignore package)) '())
+(defun package-%locally-nicknamed-by (package) (declare (ignore package)) '())
+
+(deftype string-designator () '(or character symbol string))
 
 (defun find-package (name)
-  (if (typep  name 'package)
-    name
-    (%find-pkg (string name))))
+  (check-type name (or package string-designator))
```

Comment:
I think we want to use ccl::require-type here if we want to use anything at all.

Reply:
`shka` on Freenode told me that `check-type` or a similar construct might be wanted for code clarity. I think that it would take more than this to refactor the package-related code, but I put these lines here nonetheless.

I'll modify these to use `ccl::require-type` tomorrow.

Reply:
Done.